src/com/facebook/buck/apple/AppleTest.java

```diff
@@ -265,6 +268,12 @@ public void readOutput(InputStream output) throws IOException {
     Optional<Path> testHostAppPath = extractBundlePathForBundle(testHostApp, buildContext);
     Optional<Path> uiTestTargetAppPath = extractBundlePathForBundle(uiTestTargetApp, buildContext);
 
+    ImmutableMap<String, String> testEnvironmentOverrides =
+        ImmutableMap.<String, String>builder()
+            .putAll(options.getEnvironmentOverrides())
+            .putAll(this.testSpecificEnvironmentVariables.orElse(ImmutableMap.of()))
```

Comment:
In this case a value `--test-runner-env` could be overwritten by a target specific variable in a test. I think this makes sense as target level is more granular than a setting for a buck test command. However one could also argue that it is better to have a cli value override a value defined from a file. I don't have a strong preference here.

Reply:
@robbertvanginkel I believe having it in the above order is beneficial as it allows developer control in both local and remote environments. In a local env, a developer can control both `--test-runner-env` and BUCK files. In a remote env, developers can usually only control BUCK files.